### PR TITLE
feat: TSL2561 sensor integration with lux meter

### LINKED ISSUE #1092: TSL2561 sensor integration with lux meter
**Actual Behaviour**

Built in sensor and BH1750 sensor are the sensors that can be worked with Lux meter

**Expected Behaviour**

Lux meter should have the TSL sensor to get data using PSLab device 

**Steps to reproduce it**

Go to Lux meter -> Configure tab -> Check Select sensor spinner

**LogCat for the issue**

> None

**Screenshots of the issue**

> None

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Vikum94**
Fixes #1092 

**Changes**: Adding TSL 2561 sensor functionality to the Lux meter

**Screenshot/s for the changes**: 
screenshot_2018-06-20-08-56-15

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[tls-with-lux meter.zip](https://github.com/fossasia/pslab-android/files/2117745/tls-with-lux.meter.zip)

### CODE REVIEW

**Comment by Vikum94** [comment]
@CloudyPadmal @Avjeet @harsh-2711 @abhinavraj23 Please review. Thanks

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java [hunk 1]
```diff
@@ -157,8 +160,23 @@ else if (value < updatePeriodMin)
             @Override
             public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                 try {
-                    if (bh1750 != null) {
-                        bh1750.setRange(gainValue.getSelectedItem().toString());
+                    switch (position) {
+                        case 0:
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java]: What are these blank case statements?

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java [hunk 2]
```diff
@@ -235,4 +270,18 @@ public int getValueFromText(EditText editText, int lowerBound, int upperBound) {
...
     }
+
+    public static BH1750 getBH1750() {
+        if (bh1750 != null) {
+            return bh1750;
+        }
+        return null;
+    }
+
+    public static TSL2561 getTSL2561() {
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java]: Can't we create these methods in respective classes?

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java (hunk 1 above)
Comment (Vikum94) [reply app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java]: For all cases 0, 1 and 2 do the same thing

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -851,6 +851,15 @@
     <string-array name="select_light_sensor">
         <item>Built In</item>
         <item>BH1750</item>
+        <item>TSL2561</item>
+    </string-array>
+    <string-array name="select_lux_gain">
+        <item>No gain</item>
+        <item>1x</item>
+        <item>16x</item>
+        <item>500mLx</item>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Why are these having different units? If it is a gain, shouldn't it be unit-less?

Comment (Vikum94) [reply app/src/main/res/values/strings.xml]: The previous implementation for BH1750 sensor in Sensors instrument was like this. I'll change them.

**Comment by Avjeet** [comment]
@Vikum94 what is the status with this PR?

**Comment by Vikum94** [comment]
@CloudyPadmal Bugs fixed. @Avjeet @abhinavraj23 @harsh-2711 Please review. Thanks!

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java [hunk 4]
```diff
@@ -157,8 +162,23 @@ else if (value < updatePeriodMin)
...
+                        case 0:
+                        case 1:
+                        case 2:
+                            if (bh1750 != null) {
+                                bh1750.setRange(gainValue.getSelectedItem().toString());
+                            }
+                            break;
+                        case 3:
+                        case 4:
+                        case 5:
```
Comment (abhinavraj23) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java]: @Vikum94 I think you can combine switch cases in single line eg. case 1: case 2: case 3:

Comment (Vikum94) [reply app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java]: @abhinavraj23 When file reformatted the result is this. Do you want me to change it?

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -140,6 +140,7 @@
     <string name="initialising_wait">Initialising Wait…</string>
     <string name="device_connected_successfully">Device Connected Successfully</string>
     <string name="device_not_found">Device Not Found</string>
+    <string name="sensor_not_connected">Sensor not connected</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: There are two strings with the same name "sensor_not_connected"

**Pull request merged**